pkg/gitproviders/provider.go

```diff
@@ -89,7 +91,11 @@ func deployKeyExists(ctx context.Context, repo gitprovider.UserRepository) (bool
 func uploadDeployKey(ctx context.Context, repo gitprovider.UserRepository, deployKeyInfo gitprovider.DeployKeyInfo) error {
 	_, err := repo.DeployKeys().Create(ctx, deployKeyInfo)
 	if err != nil {
-		return fmt.Errorf("error uploading deploy key %s", err)
+		if errors.Is(err, gitprovider.ErrNotFound) {
```

Comment:
in general in Go we should use else clauses unless strictly necessary, which is not the case here since we call `return` in the `if`, so a suggestion would be changing to:
```go
if errors.Is(err, gitprovider.ErrNotFound) {
	return RepositoryNoPermissionsOrDoesNotExistError
}

return fmt.Errorf("error uploading deploy key %s", err)
```